### PR TITLE
Support collections

### PR DESCRIPTION
**Pull request by Averethel**
_MINOR_

## restaurants/index

```json
[{
  "id": 1,
  "name": "Casa Ferlin"
}, {
  "id": 2,
  "name": "Le Tiguelet 
}]
```

## Before
```ruby
response = LHC.get(:restaurants/index)
response.data

# NoMethodError: undefined method `keys' for #<Array>
```

## After

```ruby
response = LHC.get(:restaurants/index)
response.data

# LHC::Response::Data ...
```

### CODE REVIEW

File: lib/lhc/response/data.rb [hunk 1]
```diff
@@ -1,36 +1,22 @@
...
-
-  attr_reader :response
-
-  def set_dynamic_accessor_methods
-    as_json.keys.each do |key|
-      define_singleton_method key do |*args|
-        as_open_struct.send key, *args
-      end
-    end
+  def method_missing(method, *args, &block)
```
Comment (spape) [inline lib/lhc/response/data.rb]: needs `responds_to_missing` implementation
https://robots.thoughtbot.com/always-define-respond-to-missing-when-overriding

File: lib/lhc/response/data.rb [hunk 2]
```diff
@@ -19,4 +19,8 @@ def initialize(response, data: nil)
   def method_missing(method, *args, &block)
     @_data.send(method, *args, &block)
   end
+
+  def respond_to_missing?(method_name, include_private = false)
+    @_data.respond_to(method_name, include_private) || super
```
Comment (spape) [inline lib/lhc/response/data.rb]: it's `respond_to?`. test. see LHS somewhere.

Comment (Averethel) [reply lib/lhc/response/data.rb]: sorry my computer is lagging so much it is actually hardly possible to do anything, must've accidentally delete the `?`. Fixed now

File: lib/lhc/response/data.rb [hunk 3]
```diff
@@ -1,36 +1,26 @@
...
-    response.format.as_json(response)
-  end
+  def initialize(response, data: nil)
+    @response = response
+    @data = data
 
-  def as_open_struct
-    response.format.as_open_struct(response)
+    if as_json.is_a?(Hash)
+      @_data = LHC::Response::Data::Item.new(response, data: data)
```
Comment (spape) [inline lib/lhc/response/data.rb]: Instance variables are already internals. Why using underscore here?
It's confusing with also dealing with `@data`.

Suggesting something like `@base` or so.

Comment (Averethel) [reply lib/lhc/response/data.rb]: Yeah, I agree - less than ideal.
`@data` is used when you want to pass data chunk extracted from the response already, will rename to `@base`

**Pull request merged**